packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -232,6 +295,66 @@ export abstract class BaseService extends Resource
     return ret;
   }
 
+  public loadBalancerTarget(options: LoadBalancerTargetOptions): IEcsLoadBalancerTarget {
+    const self = this;
+    const hostPort = this.taskDefinition._findHostPort(options);
+    return {
+      attachToApplicationTargetGroup(targetGroup: elbv2.ApplicationTargetGroup): elbv2.LoadBalancerTargetProps {
+        let portRange: ec2.Port;
+        if (hostPort.protocol === Protocol.TCP) {
+          portRange = hostPort.port === 0 ? EPHEMERAL_PORT_RANGE : ec2.Port.tcp(hostPort.port);
+        } else {
+          portRange = hostPort.port === 0 ? EPHEMERAL_PORT_RANGE : ec2.Port.udp(hostPort.port);
+        }
+        targetGroup.registerConnectable(self, portRange);
+        return self.attachToELBv2(targetGroup, options);
+      },
+
+      attachToNetworkTargetGroup(targetGroup: elbv2.NetworkTargetGroup): elbv2.LoadBalancerTargetProps {
+        return self.attachToELBv2(targetGroup, options);
+      },
+    };
+  }
+
+  public registerContainerTargets(containers: TargetProps[]) {
```

Comment:
Using `registerContainerTargets` we could potentially make some setting (e.g., target group port) opinionated so that users don't need to set up some properties. Also this is a team decision and we feel like it would make more senses for some users to start from `service` instead of `listener` or `target group`. However, we might be wrong since we aren't real users. I agree that this implementation needs some tweaks, if we decide to keep it.

Reply:
Would you be open to separating the concerns?

As in, add the fine-grained target control in this PR, and add the convenience function in a future PR? I would like to separate them out very cleanly.

If you insist on adding it, what I would like to do is mark the function and every new  interface that is used to support it `@experimental`.

Reply:
I am more inclined to put it in another PR so that I can invite my teammate to review and help refine the implementation, because we feel like this is a behavior we should support.